custom_components/resmed_myair/client/rest_client.py

```diff
@@ -0,0 +1,704 @@
+            return True
+        return False
+
+    async def _resmed_response_error_check(
+        self, step: str, response: ClientResponse, resp_dict: dict, initial: bool | None = False
+    ) -> None:
+        if "errors" in resp_dict:
+            try:
+                error_message: str = f"{resp_dict['errors'][0]['errorInfo']['errorType']}: {resp_dict['errors'][0]['errorInfo']['errorCode']}"
+                if resp_dict["errors"][0]["errorInfo"]["errorType"] == "unauthorized":
+                    if step == "ggl_query" and not initial:
+                        raise ParsingError(f"Getting unauthorized error on {step} step. {error_message}")
+                    raise AuthenticationError(
+                        f"Getting unauthorized error on {step} step. {error_message}"
+                    )
+                if (
+                    resp_dict["errors"][0]["errorInfo"]["errorType"] == "badRequest"
+                    and resp_dict["errors"][0]["errorInfo"]["errorCode"]
+                    in ("onboardingFlowInProgress", "equipmentNotAssigned")
+                ):
+                    raise IncompleteAccountError(f"{error_message}")
+            except TypeError:
+                error_message = "Error"
+                pass
+            raise HttpProcessingError(
+                code=response.status,
+                message=f"{step} step: {error_message}. {resp_dict})",
+                headers=response.headers, # type: ignore
+            )
+
+    async def _authn_check(self) -> str:
+        authn_url: str = OAUTH_URLS["authn_url"].format(
+            okta_url=self._region_config["okta_url"]
+        )
+        json_query: dict[str, Any] = {
+            "username": self._config.username,
+            "password": self._config.password,
+        }
+        _LOGGER.debug(f"[authn_check] authn_url: {authn_url}")
+        _LOGGER.debug(
+            f"[authn_check] headers: {async_redact_data(self._json_headers, KEYS_TO_REDACT)}"
+        )
+        _LOGGER.debug(
+            f"[authn_check] json_query: {async_redact_data(json_query, KEYS_TO_REDACT)}"
+        )
+
+        async with self._session.post(
+            authn_url,
+            headers=self._json_headers,
+            json=json_query,
+            cookies=self._cookies,
+        ) as authn_res:
+            _LOGGER.debug(f"[authn_check] authn_res: {authn_res}")
+            authn_dict: dict[str, Any] = await authn_res.json()
+            _LOGGER.debug(
+                f"[authn_check] authn_dict: {async_redact_data(authn_dict, KEYS_TO_REDACT)}"
+            )
+            await self._resmed_response_error_check("authn", authn_res, authn_dict)
+        if "status" not in authn_dict:
+            raise AuthenticationError("Cannot get status in authn step")
+        status: str = authn_dict["status"]
+        if status == AUTH_NEEDS_MFA:
+            if "stateToken" not in authn_dict:
+                raise AuthenticationError("Cannot get stateToken in authn step")
```

Comment:
when does this happen? This is going to be inscrutable to the user, or do we always restart the login flow after an error like this?

Reply:
In the initial setup this will be caught by the config flow and show "Invalid username or password. Please verify at myair.resmed.com" to the user.
If it happens later it will be caught by the Coordinator and trigger reauthentication.